sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/convert/MappingCosmosConverter.java

```diff
@@ -142,12 +165,83 @@ public JsonNode writeJsonNode(Object sourceEntity) {
             final String id = value == null ? null : value.toString();
             cosmosObjectNode.put("id", id);
         }
+        if (transientFields != null && !transientFields.isEmpty()) {
+            // Strip fields with @Transient annotation from the JsonNode so they are not persisted.
+            cosmosObjectNode = stripTransientFields(cosmosObjectNode, transientFields);
+        }
 
         mapVersionFieldToEtag(sourceEntity, cosmosObjectNode);
 
         return cosmosObjectNode;
     }
 
+    /**
+     * To repopulate any transient fields that were stripped from the original object
+     * @param <T> type of source entity
+     * @param transientValues transient fields and their values
+     * @param responseItem must not be {@literal null}
+     * @return CosmosItemProperties
+     * @throws MappingException no mapping metadata for entity type
+     * @throws CosmosAccessException fail to map document value
+     */
+
+    public <T> JsonNode repopulateAnyTransientFieldsFromMap(JsonNode responseItem, HashMap<Field, Object> transientValues) {
+        if (transientValues.isEmpty()) {
+            return responseItem;
+        }
+        ObjectNode updatedItem = (ObjectNode) responseItem;
+        for (Field field : transientValues.keySet()) {
```

Comment:
Instead of keySet(), you can use Map.Entry iterator for ease.

Reply:
Ok, will change it.